vyper/semantics/analysis/data_positions.py

```diff
@@ -162,20 +160,15 @@ def set_storage_slots_with_overrides(
 
         variable_name = GLOBAL_NONREENTRANT_KEY
 
-        # re-entrant key was already identified
-        if variable_name in ret:
-            continue
-
         # Expect to find this variable within the storage layout override
         if variable_name in storage_layout_overrides:
             reentrant_slot = storage_layout_overrides[variable_name]["slot"]
             # Ensure that this slot has not been used, and prevents other storage variables
             # from using the same slot
-            reserved_slots.reserve_slot_range(reentrant_slot, 1, variable_name)
+            reserved_slots.reserve_slot_range(reentrant_slot, NONREENTRANT_KEY_SIZE, variable_name)
 
             type_.set_reentrancy_key_position(VarOffset(reentrant_slot))
-
-            ret[variable_name] = {"type": "nonreentrant lock", "slot": reentrant_slot}
+            break
```

Comment:
will set the key only for the first function.
If `break` is removed, then `reserve_slot_range` will throw a `StorageLayoutException`